crates/pixi_core/src/lock_file/resolve/pypi.rs

```diff
@@ -1071,7 +1085,22 @@ async fn lock_pypi_packages(
                         }
                         SourceDist::Git(git) => {
                             // convert resolved source dist into a pinned git spec
-                            let pinned_git_spec = into_pinned_git_spec(git.clone());
+                            let mut pinned_git_spec = into_pinned_git_spec(git.clone());
+
+                            // Look up the original requirement to get the git reference (branch/tag)
+                            // It may have resolved the reference to just a commit, losing the branch/tag info
+                            if let Some(original_req) = original_requirements
+                                .iter()
+                                .find(|r| &r.name == dist.name())
+                                && let RequirementSource::Git {
+                                    git: original_git, ..
+                                } = &original_req.source
+                            {
+                                // Use the reference from the original requirement instead of what UV resolved
+                                pinned_git_spec.source.reference =
```

Comment:
Do you always want to override the reference? It may be that the reference is already set, maybe we can skip in this case, searching through the original requirements?